spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/spanner/GcpSpannerAutoConfiguration.java

```diff
@@ -142,4 +144,17 @@ public SpannerSchemaUtils spannerSchemaUtils(
 			SpannerConverter spannerConverter) {
 		return new SpannerSchemaUtils(spannerMappingContext, spannerConverter);
 	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	public SpannerDatabaseAdminTemplate spannerDatabaseAdminTemplate(
+			DatabaseAdminClient databaseAdminClient, DatabaseId databaseId) {
+		return new SpannerDatabaseAdminTemplate(databaseAdminClient, databaseId);
+	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	public DatabaseAdminClient databaseAdminClient(Spanner spanner) {
+		return spanner.getDatabaseAdminClient();
```

Comment:
Does it really make sense to be so fine-grained? 
Can `DatabaseAdminClient` be created outside of the `Spanner`?
Doesn't this show us that we can have `DatabaseAdminClient` only via calling `Spanner` API?
Thanks

Reply:
I don't think there's another way to get a db admin client, but I thought the alternative to having this bean here is to put the very broad Spanner object as the constructor argument to classes that otherwise would only need the DatabaseAdminClient and have them call getDatabaseAdminClient themselves.  I thought calling this method for them would be preferable. I could be missing something though...

Reply:
Well, my point is to inject `Spanner` to the previous `SpannerDatabaseAdminTemplate` bean definition and call this `getDatabaseAdminClient()` there for the ctor.

But I can live with this bean as well, if there is a good explanation "Why?".

Reply:
My preference to call .getDatabaseAdminClient() in this autoconfig class instead of the SpannerDatabaseAdminTemplate is that you can do much more than just db admin operations with a Spanner object, and we only want the admintemplate to have the dbadminclient.

Reply:
I'm fine with this too. So, never mind.
Just to finish a discussion and close all the gaps, let me show what I mean anyway:
```
@Bean
@ConditionalOnMissingBean
public SpannerDatabaseAdminTemplate spannerDatabaseAdminTemplate(
		Spanner spanner, DatabaseId databaseId) {
	return new SpannerDatabaseAdminTemplate(spanner.getDatabaseAdminClient(), databaseId);
}
```
That's all. No any `DatabaseAdminClient` beans requirements.